lib/_http_client.js

```diff
@@ -68,6 +68,11 @@ function ClientRequest(options, cb) {
   self.socketPath = options.socketPath;
   self.timeout = options.timeout;
 
+  const _method = options.method;
+  if (_method !== undefined && _method !== null
+    && typeof _method !== 'string') {
+    throw new TypeError('Method must be a string');
+  }
```

Comment:
In this case the `options.method` if is set to `0` results in a falsy value which is going to default the method to `GET` in the following check.